### PR TITLE
Scan N' Plan update that complies with multiple polygon selection tool

### PR DESCRIPTION
**Pull request by BryanMqz**

### CODE REVIEW

File: snp_tpp/src/roi_selection_mesh_modifier.cpp [hunk 1]
```diff
@@ -5,38 +5,45 @@
...
+
+  for(const geometry_msgs::msg::PolygonStamped& inner_vector_ : boundary_)
   {
-    Eigen::MatrixX3d boundary(boundary_.size(), 3);
-    for (Eigen::Index i = 0; i < boundary.rows(); ++i)
+    if (inner_vector_.polygon.points.size() < 5) // 5 because one single point/dot counts as 2, so a triangle (3 points in space) will count as size 6
+    {
+      std::cout << "***** Polygon points size is less than 3, size is: " << inner_vector_.polygon.points.size()/2 << " *****" << std::endl;
+      continue;
+    }
```
Comment (marip8) [inline snp_tpp/src/roi_selection_mesh_modifier.cpp:29]: Can you explain this in more detail? I don't understand why a polygon would need to have more than 3 points minimum

Comment (BryanMqz) [reply snp_tpp/src/roi_selection_mesh_modifier.cpp]: Fixed double point generation by updating rviz_polygon_selection_tool repo, where I added an rviz property tool to define the gap between the last generated point and the new one the user wants to generate.
Commit id for scan_n_plan: [1d40097](https://github.com/ros-industrial-consortium/scan_n_plan_workshop/pull/114/commits/1d4009742dacbebb3b21d117dbabd3f3cac43f77)
Commit id for rviz_polygon_selection_tool: [79744be](https://github.com/BryanMqz/rviz_polygon_selection_tool/commit/79744be1fee7e84d035e32777875cd3b654e55be)

File: snp_tpp/src/tpp_widget.cpp [hunk 2]
```diff
@@ -84,9 +84,14 @@ void TPPWidget::callback(const snp_msgs::srv::GenerateToolPaths::Request::Shared
...
-    std::vector<noether::ToolPaths> tool_paths = pipeline.plan(mesh);
+    std::vector<noether::ToolPaths> tool_paths_vector = pipeline.plan(mesh);
 
-    res->tool_paths = toMsg(tool_paths.at(0));
+    // Concatenate the tool paths from each sub-mesh into a single tool paths object
+    noether::ToolPaths tool_paths_concat;
+    for(const noether::ToolPaths& tool_paths : tool_paths_vector)
+      tool_paths_concat.insert(tool_paths_concat.end(), tool_paths.begin(), tool_paths.end());
+
+    res->tool_paths = toMsg(tool_paths_concat);
```
Comment (marip8) [inline snp_tpp/src/tpp_widget.cpp]: @marrts what do you think about merging tool paths from different selection regions into one "super raster" for the motion planner? Essentially we would be using the transition planner to plan between regions. The alternative is to modify the planning service and server to plan for multiple raster regions, which seems a bit involved and potentially unnecessary (at least for now)

Comment (marip8) [reply snp_tpp/src/tpp_widget.cpp]: Sounds like there are changes in the latest version of `tesseract` that we would like to incorporate into the motion planner configuration update, so let's merge this first

**Pull request merged**